client/components/forms/clipboard-button/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { Button } from '@automattic/components';
+import classNames from 'classnames';
+import { forwardRef } from 'react';
+import type { ButtonProps } from '@automattic/components';
+import type { ForwardRefRenderFunction } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+interface ClipboardProps extends ButtonProps {
+	text: string | ( () => string );
+	onCopy?: () => void;
+}
+
+// A button which, when clicked, copies the given text to the clipboard.
+const ClipboardButton: ForwardRefRenderFunction< HTMLButtonElement, ClipboardProps > = (
+	{ className, text, onCopy = noop, ...rest },
+	ref
+) => {
```

Comment:
I don't think that the `ButtonProps` type is compatible with the `prop` accepted by the `Button` from `@wordpress/button`

The thing is, the `Button` in `@wordpress/components` hasn't been typed yet. And so I'm not sure what's the best option here while we wait for `Button` to be typed — I know that definitely typed exposes some types, but I'm not sure if they're up to date @sirreal and @sarayourfriend do you have any insights? Thank you!

Reply:
Everything I see here is about `@automattic/components`, not `@wordpress/components`. Is that correct? If so, it's not clear to me how `@wordpress/components` is involved.

Reply:
🤦  I clearly got confused by the context (repo) switching and misread the lines of code above.

Feel free to ignore this comment, and sorry for the ping!

Reply:
it's definitely confusing to have so many button components!